Source/WebKit/UIProcess/XR/openxr/OpenXRInputMappings.h

```diff
@@ -0,0 +1,154 @@
+ * aint with this library; see the file COPYING.LIB.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#pragma once
+
+#if ENABLE(WEBXR) && USE(OPENXR)
+
+#include <array>
+#include <wtf/text/WTFString.h>
+
+namespace WebKit {
+
+using OpenXRProfileId = ASCIILiteral;
+using OpenXRButtonPath = ASCIILiteral;
+
+enum class OpenXRButtonType {
+    Trigger,
+    Squeeze,
+    Touchpad,
+    Thumbstick,
+    Thumbrest,
+    ButtonA,
+    ButtonB
+};
+
+constexpr std::array<OpenXRButtonType, 7> openXRButtonTypes {
+    OpenXRButtonType::Trigger, OpenXRButtonType::Squeeze, OpenXRButtonType::Touchpad, OpenXRButtonType::Thumbstick, OpenXRButtonType::Thumbrest,
+    OpenXRButtonType::ButtonA, OpenXRButtonType::ButtonB
+};
+
+constexpr ASCIILiteral s_pathSelect { "/input/select"_s };
+constexpr ASCIILiteral s_pathPinchExt { "/input/pinch_ext"_s };
+constexpr ASCIILiteral s_pathGraspExt { "/input/grasp_ext"_s };
+
+constexpr ASCIILiteral s_pathActionClick { "/click"_s };
+constexpr ASCIILiteral s_pathActionTouch { "/touch"_s };
+constexpr ASCIILiteral s_pathActionValue { "/value"_s };
+
+inline String buttonTypeToString(OpenXRButtonType type)
+{
+    switch (type) {
+    case OpenXRButtonType::Trigger: return "trigger"_s;
+    case OpenXRButtonType::Squeeze: return "squeeze"_s;
+    case OpenXRButtonType::Touchpad: return "touchpad"_s;
+    case OpenXRButtonType::Thumbstick: return "thumbstick"_s;
+    case OpenXRButtonType::Thumbrest: return "thumbrest"_s;
+    case OpenXRButtonType::ButtonA: return "buttona"_s;
+    case OpenXRButtonType::ButtonB: return "buttonb"_s;
+
+    default:
+        ASSERT_NOT_REACHED();
+        return emptyString();
+    }
+}
+
+enum OpenXRButtonFlags {
+    Click = 1u << 0,
+    Touch = 1u << 1,
+    Value  = 1u << 2,
+};
+
+enum class OpenXRHandFlags {
+    Left = 1u << 0,
+    Right = 1u << 1,
+    Both = Left | Right
+};
+
+struct OpenXRButton {
+    OpenXRButtonType type;
+    OpenXRButtonPath path;
+    OpenXRButtonFlags flags;
+    OpenXRHandFlags hand;
+};
+
+enum class OpenXRAxisType {
+    Touchpad,
+    Thumbstick
+};
+
+constexpr std::array<OpenXRAxisType, 2> openXRAxisTypes {
+    OpenXRAxisType::Touchpad, OpenXRAxisType::Thumbstick
+};
+
+inline String axisTypetoString(OpenXRAxisType type)
+{
+    switch (type) {
+    case OpenXRAxisType::Touchpad: return "touchpad"_s;
+    case OpenXRAxisType::Thumbstick: return "thumbstick"_s;
+    default:
+        ASSERT_NOT_REACHED();
+        return emptyString();
+    }
+}
+
+struct OpenXRAxis {
+    OpenXRAxisType type;
+    OpenXRButtonPath path;
+};
+
+struct OpenXRInteractionProfile {
+    ASCIILiteral path;
+    std::span<const OpenXRProfileId> profileIds;
+    std::span<const OpenXRButton> buttons;
+    std::span<const OpenXRAxis> axes;
+};
+
+constexpr std::array<OpenXRProfileId, 3> handInteractionProfileIds { "generic-hand-select-grasp", "generic-hand-select", "generic-hand" };
```

Comment:
```suggestion
constexpr std::array<OpenXRProfileId, 3> handInteractionProfileIds { "generic-hand-select-grasp"_s, "generic-hand-select"_s, "generic-hand"_s };
```

or is it there an exception in this case?

Reply:
not needed apparently